### DISCUSSION TITLE
Does Flecs.Net support Android(arm64-v8a) platform now?

### DISCUSSION THREAD

**Discussion by isJDongYa** [BeanCheeseBurrito/Flecs.NET#78] [Q&A]
I directly reference it in godot c# project .csporj file, and then export it to android(arm64-v8a)

But when I create a "World" use World.Create(), it emits can't find "flecs" dll error.

I just wan to know it this project support compile directly to android, or what i can do to support it?

**Comment by BeanCheeseBurrito** [comment]
Flecs.NET does not currently build native libraries for Android.

If you would like to try to add support for Android, you can look into modifying the zig build script to handle Android targets.
https://github.com/BeanCheeseBurrito/Flecs.NET/blob/main/src/Flecs.NET.Native/build.zig
https://github.com/BeanCheeseBurrito/Flecs.NET/blob/main/src/Flecs.NET.Native/Flecs.NET.Native.csproj

In the zig script, you need to add a new switch case for Android platforms and include the correct paths to the Android SDK's ``/usr/include`` and ``/usr/lib`` paths. The code below shows how this is done for IOS native libraries.
https://github.com/BeanCheeseBurrito/Flecs.NET/blob/caddc4cf12c28c58bd29c3d36beb38eff78965bd/src/Flecs.NET.Native/build.zig#L163-L170

The .csproj then needs to be updated to pass Android-specific build arguments to zig like below.

https://github.com/BeanCheeseBurrito/Flecs.NET/blob/caddc4cf12c28c58bd29c3d36beb38eff78965bd/src/Flecs.NET.Native/Flecs.NET.Native.csproj#L120-L125

An example of what the above might look like for Android.
```xml
<When Condition="$(RuntimeIdentifier) == 'android-arm64'">
    <PropertyGroup>
        <ZigIdentifier>aarch64-linux-android</ZigIdentifier>
        <ZigArgs>--sysroot $(ANDROID_SDK)</ZigArgs>
    </PropertyGroup>
</When>
```

**Reply by isJDongYa** [reply]
I try to compile it for android as your advices.

I change build.zig in Flecs.Net.Native like this
![a0bcef8159538e54fd430d795e1e9683](https://github.com/user-attachments/assets/e4c7a15f-bed0-4f16-a713-83a5a36c4d1a)

and change the .csproj like this
![image](https://github.com/user-attachments/assets/b75897d2-30b9-4045-99fe-1f6dccd8d9d0)

My andorid sdk places at "C:/Users/Jidon/AppData/Local/Android/Sdk"

But when I run command: dotnet build --runtime "android-arm64"，it emit error:
![image](https://github.com/user-attachments/assets/b42e0fd3-cbec-4d11-9a30-f86340281ed7)

Error: "Libc is not available"

Then I run "zig targets > targets.json"

The file show that zig doesn't support "aarch64-linux-android"  libc.

I am not familiar with zig, and I google search the problem, but I can't find any way to solve it.
What can i do for it?

**Reply by BeanCheeseBurrito** [reply]
That was the same error I ran into the last time I tried to add support for Android builds. I don't know what the fix is unfortunately.
